### PR TITLE
Definition @API interface

### PR DESCRIPTION
**Pull request by cos**
- adds an `API` module which will be an easy entrypoint for third-parties into the K project -- if they want more control, they can still do injection, etc. instead
- adds a `Definition` class for dealing with loading `Definition`s.

### CODE REVIEW

**Comment by cos** [comment]
Jenkins: test this please.

**Comment by cos** [comment]
I'm not sure about using another class named "Definition". I could also move this to `org.kframework.Parser` and have `loadDefinition` methods there.

**Comment by cos** [comment]
@dwightguth, I'm adding a module so you may also want to have a look.

**Comment by dwightguth** [comment]
this looks okay to me but bear in mind that so long as you aren't able to set more advanced flags on the parsing process, this is not going to be able to completely replace calls to kompile.

**Comment by cos** [comment]
Yes, it's definitely far from a replacement of kompile as it has only very limited configurability. I'm trying to retrain myself from adding configurable inputs. Only adding them as needed.

**Comment by cos** [comment]
@daejunpark, could you have a look at this? Mostly refactoring and adding another @API interface.

File: kernel/src/main/java/org/kframework/Definition.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+@API
+public class Definition {
+
+    /**
+     * Parses the text to create a {@link Definition} object.
+     * The main module of the definition will be last module defined in the text file.
+     */
+    public static org.kframework.definition.Definition from(String definitionText) {
+        Pattern pattern = Pattern.compile("module ([A-Z_]*)");
```
Comment (andreistefanescu) [inline kernel/src/main/java/org/kframework/Definition.java]: Could this interfere in unexpected ways with the contents of the module, e.g. some syntax names `foo_module`?

Comment (cos) [inline kernel/src/main/java/org/kframework/Definition.java]: it might... I'll try to restrict it a bit

Comment (cos) [inline kernel/src/main/java/org/kframework/Definition.java]: the cleanest way would be to not require a "main" module, and only select one when need, but I'm not going into that large a change now.

**Comment by andreistefanescu** [comment]
lgtm

**Comment by cos** [comment]
Jenkins: test this please.

**Comment by cos** [comment]
made the regex a bit more specific. merge if it ok.

**Pull request merged**